File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        windows = []
        max_length = 0

        for right in range(len(s)):
            while s[right] in windows:
                windows.pop(0)

            windows.append(s[right])

            max_length = max(max_length, len(windows))

        return max_length

        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.dart
class Solution {
  int lengthOfLongestSubstring(String s) {
       Set<String> charSet = {};
       int left = 0;
       int maxWindow = 0;

       for(int right = 0; right < s.length ; right++) {
           while(charSet.contains(s[right])) {
              charSet.remove(s[left]);
              left++;
           }
        charSet.add(s[right]);
        maxWindow = max(maxWindow, right - left + 1);   
       }
    return maxWindow;   
  }
}